### PR TITLE
Add recorder vars db_max_retries and db_retry_wait

### PR DESCRIPTION
**Pull request by TechnicallyJoe**
## Proposed change
Essentially, i'm proposing to add:

    db_retry_wait:
    db_max_retries:

to the recorder component. These are necessary because currently the retry count and sleep time is fixed to 10 retries, and 3 seconds sleep. I want to change that, so that its a variable.

The need came from an issue i had where my mariadb were consistently so slow to start, that my Home assistant container made its 10 retries and moved on. Testing shows that a completely new mariadb is about 2 mins to start up, so i was trying to figure out a way to force HA to check a bit more.
Proposed fixed were scripting the deploy, wrapping Home Assistant in a wait-for-it or dockerize container and using that to wait, but this seems like the best solution that benefits most people.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/home-assistant/pull/31561
- This PR fixes or closes issue: none.

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_integrations/recorder.markdown
Comment (springstan) [inline source/_integrations/recorder.markdown]:
```suggestion
      type: integer
```

File: source/_integrations/recorder.markdown
Comment (springstan) [inline source/_integrations/recorder.markdown]:
```suggestion
      type: integer
```

File: source/_integrations/recorder.markdown
Comment (springstan) [inline source/_integrations/recorder.markdown]: Please do not remove this statement.

File: source/_integrations/recorder.markdown
Comment (TechnicallyJoe) [reply source/_integrations/recorder.markdown]: Fixed.

File: source/_integrations/recorder.markdown
Comment (TechnicallyJoe) [reply source/_integrations/recorder.markdown]: No idea how that happened. Very sorry! - Readded.

File: source/_integrations/recorder.markdown
Comment (TechnicallyJoe) [reply source/_integrations/recorder.markdown]: fixed.

**Review by springstan — APPROVED** [review]
Looks good to me 👍 however someone from the documentation team should approve it as well.

**Review by klaasnicolaas — APPROVED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged 👍

**Pull request merged**
